mrp_merge_productions/mrp_production.py

```diff
@@ -0,0 +1,320 @@
+                _("Unable to merge 'To Produce' move"),
+                _("No candidate to merge move %s") % (to_merge.id, ),
+            )
+
+
+class mrp_production(orm.Model):
+    _inherit = 'mrp.production'
+
+    def _check_productions_mergeable(self, productions):
+        if any(production.state in ('done', 'cancel', 'in_production')
+               for production in productions):
+            raise orm.except_orm(
+                _("Invalid Selection"),
+                _("You cannot merge manufacturing orders that are done, "
+                  "in production or cancelled."),
+            )
+
+        unique_attrs = (
+            ("product_id", _("Product")),
+            ("bom_id", _("Bill of Materials")),
+            ("product_uom", _("Product UOM")),
+            ("location_src_id", _("Source Location")),
+            ("location_dest_id", _("Destination Location")),
+        )
+        for attr, name in unique_attrs:
+            values = set(
+                id_of(getattr(prod, attr))
+                for prod in productions
+            )
+
+            if len(values) > 1:
+                value_names = ",\n".join([
+                    name_of(value)
+                    for value in set(getattr(prod, attr)
+                                     for prod in productions)
+                ])
+                raise orm.except_orm(
+                    _("Incompatible Productions"),
+                    _("Unable to merge productions with different %s\n"
+                      "Received values:\n%s") % (name, value_names),
+                )
+
+    def _do_merge_productions(self, cr, uid, target, productions, context):
+        move_obj = self.pool["stock.move"]
+        prod_line_obj = self.pool["mrp.production.product.line"]
+        workcenter_obj = self.pool["mrp.production.workcenter.line"]
+
+        # Update quantities to produce
+        target_write = {
+            "product_qty": target.product_qty + sum(
+                prod.product_qty for prod in productions,
+            ),
+        }
+        if target.product_uos_qty:
+            target_write["product_uos_qty"] = target.product_uos_qty + sum(
+                prod.product_uos_qty for prod in productions,
+            )
+
+        _logger.debug("Writing to production %d: %r", target.id, target_write)
+        target.write(target_write)
+
+        new_picking = {
+            "picking_id": target.picking_id and target.picking_id.id,
+        }
+
+        rm_picking_ids = []
+        rm_prod_ids = []
+
+        moves_to_produce = [move for move in target.move_created_ids]
+        moves_to_consume = [move for move in target.move_lines]
+        scheduled_lines = [line for line in target.product_lines]
+        moves_to_link2 = []
+        moves_to_cancel = []
+        workcenter_lines = []
+        move_updates = {}
+        prodlines_to_delete = []
+        prodline_updates = {}
+        # XXX Should we merge notes or add notes of any kind?
```

Comment:
Yes merging notes probably makes sense, but duplicate strings should be avoided.
I don't think adding notes is wanted, but writing to the document's chat/wall could certainly be useful.
That's not a blocking point however